crates/node/src/lib.rs

```diff
@@ -205,32 +223,46 @@ impl Node {
                     .await
             })
             .await
-            .map_err(|err| Error::Attachment(cluster_controller_endpoint, err))?;
+            .map_err(|err| Error::Attachment(cluster_controller_address, err))?;
 
         Ok(())
     }
-}
-
-/// Specifying where the cluster controller runs. Options are:
-///
-/// * Local: Spawning the controller as part of this process
-/// * Remote: The controller runs on a remote host
-#[derive(Debug)]
-pub enum ClusterControllerLocation {
-    Local,
-    Remote(String),
-}
-
-impl FromStr for ClusterControllerLocation {
-    type Err = Infallible;
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let result = if s.to_lowercase() == "local" {
-            ClusterControllerLocation::Local
-        } else {
-            ClusterControllerLocation::Remote(s.to_string())
+    fn create_channel_from_network_address(
+        cluster_controller_address: &NetworkAddress,
+    ) -> Result<Channel, http::Error> {
+        let channel = match cluster_controller_address {
+            NetworkAddress::Uds(uds_path) => {
+                let uds_path = uds_path.clone();
+                Endpoint::try_from("/")
+                    .expect("/ should be a valid Uri")
+                    .connect_with_connector_lazy(service_fn(move |_: Uri| {
+                        UnixStream::connect(uds_path.clone())
+                    }))
+            }
+            NetworkAddress::TcpSocketAddr(socket_addr) => {
+                let uri = Self::create_uri(socket_addr)?;
+                Self::create_lazy_channel_from_uri(uri)
+            }
+            NetworkAddress::DnsName(dns_name) => {
+                let uri = Self::create_uri(dns_name)?;
+                Self::create_lazy_channel_from_uri(uri)
+            }
         };
+        Ok(channel)
+    }
+
+    fn create_uri(authority: impl ToString) -> Result<Uri, http::Error> {
+        Uri::builder()
+            .scheme("http")
```

Comment:
A todo that we need to make this configurable?

Reply:
Yes, will add it.